### PR TITLE
Fix test_gl error reporting

### PR DESCRIPTION
**Pull request by DziubanMaciejIntel**
Overwriting 'error' variable after check for Msaa/Depth support was clearing
the error counter incremented after failure in test_image_format_write. In
effect the test might return 0 even if there were errors.

### CODE REVIEW

File: test_conformance/gl/test_images_write_common.cpp [hunk 1]
```diff
@@ -813,15 +813,15 @@ int test_images_write_common(cl_device_id device, cl_context context,
             get_base_gl_target(targets[ tidx ]) == GL_TEXTURE_2D_MULTISAMPLE_ARRAY)
         {
             bool supports_msaa;
-            error = supportsMsaa(context, &supports_msaa);
+            bool error = supportsMsaa(context, &supports_msaa);
```
Comment (kpet) [inline test_conformance/gl/test_images_write_common.cpp:816]: Declaring another variable with the same name will shadow the previous declaration so previously recorded errors won't be tested in the condition below. I'm not sure I understand what's happening and how this fixes the issue.

Comment (DziubanMaciejIntel) [reply test_conformance/gl/test_images_write_common.cpp:816]: Yes, I meant to shadow it. IMO the test in line 825 should only test the result of supportsMsaa function. Outer scope `error` variable is used for counting how many formats failed (line 850). Suppose the scenario:
- check msaa support for first format - success, error=0
- first format validation fails - error=1
- check msaa support for second format - success, error=0
- second format validation succeds - error=0
- rest of the formats also pass

So the whole test returns 0, even though the first format validation failed.

Comment (kpet) [reply test_conformance/gl/test_images_write_common.cpp:816]: Right, the test isn't designed to stop on the first outer scope error, makes sense. I suggest you rename the inner variable to make this easier to follow and remove the unnecessary bool/int implicit conversions.

**Comment by kpet** [comment]
@DziubanMaciejIntel This needs formating fixes to go in.

**Comment by kpet** [comment]
@DziubanMaciejIntel Thanks but you missed the spaces after the if :). I suggest you use `git-clang-format` to do the formating (see https://github.com/KhronosGroup/OpenCL-CTS/blob/master/check-format.sh for an example), much easier than doing it manually.

**Comment by DziubanMaciejIntel** [comment]
@kpet right, I put too much faith in myself :D Thanks and sorry for notification spam.

**Pull request merged**